src/av2/evaluation/scene_flow/eval.py

```diff
@@ -0,0 +1,311 @@
+
+import argparse
+from functools import partial
+from pathlib import Path
+from typing import Dict, Final, Iterator, List, Optional, Tuple, Union
+
+import constants
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn.functional as F
+from rich.progress import track
+
+from av2.utils.typing import NDArrayBool, NDArrayFloat, NDArrayInt
+
+EPS: Final = 1e-10
+
+
+def epe(dts: NDArrayFloat, gts: NDArrayFloat) -> NDArrayFloat:
+    """Compute the end-point-error between predictions and ground truth.
+
+    Args:
+        dts: (N,3) array containing predicted flows
+        gts: (N,3) array containing ground truth flows
+
+    Returns:
+        The point-wise end-point-error
+    """
+    epe: NDArrayFloat = np.linalg.norm(dts - gts, axis=-1).astype(np.float64)
+    return epe
+
+
+def accuracy(dts: NDArrayFloat, gts: NDArrayFloat, threshold: float) -> NDArrayFloat:
+    """Compute the percent of inliers for a given threshold for a set of dtsictions and ground truth vectors.
+
+    Args:
+        dts: (N,3) Array containing predicted flows.
+        gts: (N,3) Array containing ground truth flows.
+        threshold: Threshold to use for classifying inliers.
+
+    Returns:
+        The pointwise inlier assignments.
+    """
+    l2_norm = np.linalg.norm(dts - gts, axis=-1)
+    gts_norm = np.linalg.norm(gts, axis=-1)
+    relative_err = l2_norm / (gts_norm + EPS)
+    error_lt_5 = (l2_norm < threshold).astype(bool)
+    relative_err_lt_5 = (relative_err < threshold).astype(bool)
+    acc: NDArrayFloat = np.logical_or(error_lt_5, relative_err_lt_5).astype(np.float64)
+    return acc
+
+
+def accuracy_strict(dts: NDArrayFloat, gts: NDArrayFloat) -> NDArrayFloat:
+    """Compute the acccuracy with a 0.05 threshold.
+
+    Args:
+        dts: (N,3) Array containing predicted flows.
+        gts: (N,3) Array containing ground truth flows.
+
+    Returns:
+        The pointwise inlier assignments at a 0.05 threshold
+    """
+    return accuracy(dts, gts, 0.05)
+
+
+def accuracy_relax(dts: NDArrayFloat, gts: NDArrayFloat) -> NDArrayFloat:
+    """Compute the acccuracy with a 0.1 threshold.
+
+    Args:
+        dts: (N,3) Array containing predicted flows.
+        gts: (N,3) Array containing ground truth flows.
+
+    Returns:
+        The pointwise inlier assignments at a 0.1 threshold.
+    """
+    return accuracy(dts, gts, 0.10)
+
+
+def angle_error(dts: NDArrayFloat, gts: NDArrayFloat) -> NDArrayFloat:
+    """Compute the angle error between dtsicted and ground truth flow vectors.
+
+    Args:
+        dts: (N,3) Array containing predicted flows.
+        gts: (N,3) Array containing ground truth flows.
+
+    Returns:
+        The pointwise angle errors
+    """
+    unit_label = gts / (np.linalg.norm(gts, axis=-1, keepdims=True) + EPS)
+    unit_dts = dts / (np.linalg.norm(dts, axis=-1, keepdims=True) + EPS)
+    dot_product = np.clip(np.sum(unit_label * unit_dts, axis=-1), a_min=-1 + EPS, a_max=1 - EPS)
```

Comment:
This is some old code that's used in a lot of repo's for computing the angle error. I assume its there because floating point errors can make the results outside the range [-1, 1]